### PR TITLE
feat: Allow running without config specified

### LINKED ISSUE #1771: Allow running without config specified
**Is your feature request related to a problem? Please describe.**
After we publish binary files along with release, it might be inconvenient to run qdrant with, as it always require config file to be present.

**Describe the solution you'd like**
Move all default values into the Rust source and allow running qdrant without config specified.
It should log the warning, at the start, that the default config is used.

If `RUN_MODE` env is specified, absence of configuration file should not be ignore

### PR DESCRIPTION
**Pull request by paulotten**
Closes #1771

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally using `cargo fmt` command prior to submission?
3. [x] Have you checked your code using `cargo clippy` command?

### CODE REVIEW

**Comment by paulotten** [comment]
Two tests failing locally. Investigating.

**Comment by paulotten** [comment]
> Two tests failing locally. Investigating.

Fixed. `test_default_config()` was setting `RUN_MODE` to "TEST", which is invalid.

**Comment by paulotten** [comment]
@generall 

It looks like I need to update openapi specifications (and probably gRPC .proto files). Does this sound right?

Before I go making changes there, does what I've changed so far look good?

File: lib/collection/src/optimizers_builder.rs [hunk 1]
```diff
@@ -57,14 +60,56 @@ pub struct OptimizersConfig {
...
+            max_segment_size: Option::default(),
+            memmap_threshold: Option::default(),
+            indexing_threshold: default_indexing_threshold(),
+            flush_interval_sec: default_flush_interval_sec(),
+            max_optimization_threads: default_max_optimization_threads(),
+        }
+    }
+}
+
+fn default_deleted_threshold() -> f64 {
```
Comment (timvisee) [inline lib/collection/src/optimizers_builder.rs:89]: `const fn` + 5 others below

File: src/settings.rs [hunk 2]
```diff
@@ -27,7 +33,41 @@ pub struct ServiceConfig {
...
+            max_workers: Option::default(),
+            enable_cors: default_service_cors(),
+            enable_tls: bool::default(),
+            verify_https_client_certificate: bool::default(),
+            api_key: Option::default(),
+        }
+    }
+}
+
+fn default_service_host() -> String {
```
Comment (timvisee) [inline src/settings.rs:54]: `const fn` + 3 others below

File: src/settings.rs [hunk 3]
```diff
@@ -183,13 +252,39 @@ fn default_message_timeout_tics() -> u64 {
     10
 }
 
+fn default_p2p_port() -> Option<u16> {
```
Comment (timvisee) [inline src/settings.rs:255]: `const fn` + 3 others below

**Review by timvisee — CHANGES_REQUESTED** [review]
Thanks for your contribution!

No logic problems. I do have some nitpicks though, please see my comments.

Could you also [update](https://github.com/qdrant/qdrant/blob/master/docs/DEVELOPMENT.md#rest) the OpenAPI specification to apply these changes?

**Comment by timvisee** [comment]
Before continuing work on this, please see: https://github.com/qdrant/qdrant/issues/1771#issuecomment-1519627674

**Comment by timvisee** [comment]
Closing this because <https://github.com/qdrant/qdrant/pull/1776> is merged (which is an alternative).

Feel free to open this again if you think there is a good reason for it.

**Pull request closed without merging**